Merge processing pipelines (#22)
sct-pipeline/contrast-agnostic-softseg-spinalcord: README.md

* Generate mean image after motion correction
* Remove intermidiate files

> Next steps are to generate a constrast-agnostic soft segmentation: 

Reviewer: ```suggestion
Next steps are to generate a contrast-agnostic soft segmentation: 
```

- creates spinal cord segmentation (if it doesn't exist yet)
- creates a mask around the T2 spinal cord
- co-register all contrasts to the T2 spinal cord 
- average all segmentations from each contrast within the same space (the T2)
- bring back the segmentations to the original image space of each contrast (except for the T2)

> - Crop images, segmentations and soft segmentation arround the spinal cord

Reviewer: ```suggestion
- Crop images, segmentations and soft segmentation around the spinal cord
```

> The output of this script is a new 'derivatives/labels_softseg/' folder that contains the soft labels to be used in this contrast-agnostic segmentation project. All the registration were manually QC-ed and the problematic registrations were listed in [exclude_reg.yml](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/main/exclude_reg.yml). The processing was run again to generate the soft segmentations. 

Reviewer: ```suggestion
The output of this script is a new `derivatives/labels_softseg/` folder that contains the soft labels to be used in this contrast-agnostic segmentation project. All the registration were manually QC-ed and the problematic registrations were listed in [exclude_reg.yml](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/main/exclude_reg.yml). The processing was run again to generate the soft segmentations. 
```

Specify the path of preprocessed dataset with the flag `-path-data`. 

### Launch processing

```
sct_run_batch -jobs -1 -path-data <PATH_DATA> -path-output <PATH-OUTPUT> -script process_data.sh -script-args exclude_reg.yml
```

A `process_data_clean` folder is created in <PATH-OUTPUT> where the cropped data and derivatives are included. Here, only the images that have a manual segmentation and soft segmentation are transfered.

### Quality control

After running the analysis, check your Quality Control (qc) report by opening the file <PATH-OUTPUT>/qc/index.html. Use the "search" feature of the QC report to quickly jump to segmentations or labeling issues.

**1. Segmentations**

If segmentation issues are noticed while checking the quality report, proceed to manual correction using the procedure below:

* In QC report, search for "deepseg" to only display results of spinal cord segmentation
* Review segmentation and spinal cord.
* Click on the F key to indicate if the segmentation is OK ✅, needs manual correction ❌ or if the data is not usable ⚠️ (artifact). Two .yml lists, one for manual corrections and one for the unusable data, will automatically be generated.

> * Download the lists by clicking on **Download QC Fails*** and on **Download Qc Artifacts**.

Reviewer: ```suggestion
* Download the lists by clicking on **Download QC Fails*** and on **Download QC Artifacts**.
```

Proceed to manual correction using FSLeyes or ITK snap. Uplaod the manual segmentations (_seg-manual.nii.gz) with json sidecar in the derivatives.
Re-run the analysis: [Launch processing](#launch-processing)

**2. Registrations**

* In QC report, search for "sct_register_multimodal" to only display results of registration.
* Click on the F key to indicate if the registration is OK ✅, needs to be excluded ❌ or if the data is not usable ⚠️ (artifact). Two .yml lists, will automatically be generated.

> * Downlaod the list by clicking on **Download QC Fails** and add the file names under `FILES_REG` to https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/main/exclude_reg.yml

Reviewer: ```suggestion
* Download the list by clicking on **Download QC Fails** and add the file names under `FILES_REG` to https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/main/exclude_reg.yml
```

* Re-run the analysis: [Launch processing](#launch-processing)

**3. Soft segmentations**

* In QC report, search for "softseg" to only display results of spinal cord soft segmentation
* Click on the F key to indicate if the soft segmentation is OK ✅, needs manual correction ❌ or if the data is not usable ⚠️ (artifact). Two .yml lists, one for manual corrections and one for the unusable data, will automatically be generated.

> * Downlaod the list by clicking on **Download QC Fails**.

Reviewer: ```suggestion
* Download the list by clicking on **Download QC Fails**.
```

> * Uplaod the soft segmentations under `data-multi-subject/darivatives/labels_softseg`.

Reviewer: ```suggestion
* Upload the soft segmentations under `data-multi-subject/derivatives/labels_softseg`.
```

write 100 times 'upload/download' on the black board 😉

Author: it was more of a copy/paste error 😂
